internal/aad/jwt.go

```diff
@@ -200,11 +227,14 @@ func (c *TokenProviderConfiguration) NewServicePrincipalToken() (*adal.ServicePr
 	return spToken, nil
 }
 
-func (c *TokenProviderConfiguration) getTokenProvider(msiEndpoint string) (*adal.ServicePrincipalToken, error){
-	if c.ClientID == "" {
-		return adal.NewServicePrincipalTokenFromMSI(msiEndpoint, c.ResourceURI)
+func (c *TokenProviderConfiguration) getTokenProvider(msiEndpoint string) (*adal.ServicePrincipalToken, error) {
```

Comment:
if we have clientID use that.
then if we have resourceID use that
then if we have nothing, assume we are using SystemAssigned